### ISSUE TITLE
batch baking isn't working right

### ISSUE THREAD

**Issue by laurentopia** [guycalledfrank/bakery-issues#102]
solo bake:
![image](https://user-images.githubusercontent.com/26075468/119892081-92d72800-beee-11eb-8266-d6c539b9a8c0.png)
batched:
![image](https://user-images.githubusercontent.com/26075468/119892110-9ec2ea00-beee-11eb-9b41-e7c94ec7e9ab.png)

i did a batch after defaulting setting then saving all the scenes to what i had saved with "save to default"

**Comment by laurentopia** [comment]
another bug: even with bake reflection probe turned on it doesn't bake the refprobes

**Comment by guycalledfrank** [comment]
Any way I can reproduce it?

**Comment by laurentopia** [comment]
I guess you're not getting that? It's always happened in this configuration.
unity 2020.1.17
urp 8.31
latest bakery, dunno the version though
![image](https://user-images.githubusercontent.com/26075468/120077200-0d728580-c05e-11eb-91e0-4e6b58728e6f.png)

**Comment by guycalledfrank** [comment]
Did a quick test with the closest setup at hand (Unity 2020.2.7, URP 10.3.2):
- Imported Bakery
- Upgraded project materials to URP
- Created a reflection probe in example_lights and example_sponza_day
- Also added some light probes to example_lights
- Added both scenes to the batch baker
- Baked

Everything worked correctly in my case:

![image](https://user-images.githubusercontent.com/7008423/120096821-3e25ee00-c136-11eb-97ee-dc2c17949e18.png)


So perhaps there is something different about your project. If you can send me any test scenes where I can reproduce this, I'll test them.

**Comment by laurentopia** [comment]
exporting a single scene is usually difficult in unity after a project gets a certain age, let's try anyway
https://we.tl/t-fujyi15VDv

**Comment by guycalledfrank** [comment]
Testing on 2020.1.14, URP 8.3.1.

Not exactly sure what it was supposed to look like:

![image](https://user-images.githubusercontent.com/7008423/121879409-7a9c4100-cd15-11eb-8cae-4c785afb8023.png)

Added some materials to everything:

![image](https://user-images.githubusercontent.com/7008423/121880990-5477a080-cd17-11eb-87e6-325071511f39.png)

Baked and got some crap due to low resolution + OptiX5, increased the resolution a bit, switched to OpenImageDenoise, baked and got this:

![image](https://user-images.githubusercontent.com/7008423/121880548-cef3f080-cd16-11eb-8214-7c7c913bb2de.png)

Created another scene. Ran batch bake with your scene. Batch bake bakes the lightmaps, but then fails at light probes, because there are no light probe groups. Additionally it destroys the lightmaps after the attempt?! :) 
Fixed:
https://github.com/guycalledfrank/bakery-csharp/commit/6be82a90350360829f1520ebac7f2840680d9fb0
https://github.com/guycalledfrank/bakery-csharp/commit/b061b8b7eabc63c84c5387c1d1a668fdab00dce7

It does bake the reflection probes though, I can't see any fails

**Comment by laurentopia** [comment]
weird on the refprobes, alright I'll take the fix :)

**Comment by mawkews** [comment]
I just ran into this issue, it seems to happen with many light meshes and a large map 
bakery: v1.98 with [patch](https://github.com/guycalledfrank/bakery-csharp/commit/954090fdf9c7c15fd61333af7277c0cd69f58b62)
unity: 2022.3.22f1

here are the settings I used and my lights in a prefab
[settings_lights.zip](https://github.com/user-attachments/files/31091278/settings_lights.zip)

**Comment by guycalledfrank** [comment]
@mawkews how do you reproduce it? Bake any random scene with these Light Meshes via the batch bake? Does it happen if it's the only scene in the batch baker list? The first scene? 2nd/last?

**Comment by mawkews** [comment]
I think made a mistake and my issue might be about something else, when I was looking at the first two images above my issue looked exactly like it. I only bake for a single scene and when I have Light Mesh batch limit set to 0 it bakes fine but when I set it to anything else it bakes lights really bright like here [image](https://user-images.githubusercontent.com/26075468/119892110-9ec2ea00-beee-11eb-9b41-e7c94ec7e9ab.png)

I was trying to reproduce it in a small scene with the same lights because my bakes take a really long time but that didn't work so I assume its those lights plus a really large scene, I was also thinking maybe its a 1080ti gpu issue

**Comment by mawkews** [comment]
I did some more testing and if I merge the lights together in blender into larger meshes it also produces a really bright bake even when Mesh batch limit is set to 0

**Comment by guycalledfrank** [comment]
@mawkews  Interesting. Batch limit does the same thing as you did in blender essentially. I can't reproduce it on a random scene with your light prefab, but since batched lights have possibly different sample distributions... it could be some weird chain of events, such as a small surface not having proper UV2 and covering a large portion of the lightmap instead; or a large surface being a tiny dot in the UV2... and then a light randomly shines on it, and it bounces off? Does the bug happen with 0 GI bounces?

Also, any particular reason you're not using self-shadow and samples near > 0? It's much more consistent with/without the batch limit (you'll need to enable Cast Shadows on the renderers though, or it won't work).

**Comment by mawkews** [comment]
Yeah I realized its essentially the same but I also noticed if I have a large light in a room and a small light in another room nearby then the large light bakes fine but the small room gets that really bright bake so it might be that since they are batched it blasts all the light in that small room? so I just broke up my lights until it stopped creating bright spots. 

For self shadow I don't remember if it was working or not I will have to try it again but the reason I have been doing large samples is because I had an issue where walls in my rooms especially in hallways had dark shadow spots and I read somewhere you commented large samples will fix that so I tried it and it worked

**Comment by mawkews** [comment]
I tried with self shadows on and everything bakes really dark and thats the only setting I changed so Im not sure what is happening there

**Comment by guycalledfrank** [comment]
Did you enable Cast Shadows on the lights though as I mentioned in the previous post?

**Comment by mawkews** [comment]
I did forget to, but after fixing it and turning shadows on it bakes large dark shadows under the lights so its only useful for lights that stay enabled but all my lights are hidden at runtime

**Comment by guycalledfrank** [comment]
Not quite sure, but can you try replacing these files in Asssets/Editor/x64/Bakery/optix9? [optix9_areachange.zip](https://github.com/user-attachments/files/31615749/optix9_areachange.zip)

(assuming you're running OptiX9 version of Bakery, which is the case if your driver version is >= 590)

**Comment by mawkews** [comment]
my driver is 582.66 currently and I was using optix 7
I believe its the latest for 1080ti

**Comment by guycalledfrank** [comment]
I mean the lightmapper backend, not the denoiser. Select ftGlobalStorage.asset and disable runsRTX6, enable runsRTX9; then recompile any script or enter/exit playmode. It should use the 9 version then.

**Comment by mawkews** [comment]
ohhhh. I tried it out but still the same issues

<img width="717" height="499" alt="Image" src="https://github.com/user-attachments/assets/edc2ac24-df23-47ef-b574-999b50ab58c7" />
<img width="634" height="521" alt="Image" src="https://github.com/user-attachments/assets/c2a68a98-01a8-4000-8575-c324431e5d1b" />

**Comment by guycalledfrank** [comment]
OK, I think I found the culprit. Fixed: https://github.com/guycalledfrank/bakery-csharp/commit/480bb150afc10151a78161f34d568b59ee151091

Can you try it with the latest patch again? 
I also wonder if it's enough with just the latest patch or you'll need the files from the archive above too.

Batching will batch less now though. The original approach didn't work correctly for LightMeshes with noticeably different sample density. So now it'll still batch together near-identical lights (matching area/samples), but separate other lights.
It's either that or I'll have to make the general case slower by storing more values per sample or exploding the already existing shader permutation explosion, and I'd rather test the easier option for now...

**Comment by mawkews** [comment]
<img width="1385" height="899" alt="Image" src="https://github.com/user-attachments/assets/621faccd-c17d-423b-8fb7-7147061c728e" />I tried the patch and it worked somewhat but now a different place has the issue, did patching overwrite the earlier files I replaced earlier? Also im wondering if the cutoff value has any effect since thats the only difference between lights, all my samples values are the same.

**Comment by mawkews** [comment]
I tried replacing the files again from optix9 + latest patch and it seems to have almost fixed it, there is one spot where I have 6 identical lights in 6 small rooms next to each other and each room gets a bit brighter with the last room hitting that same really bright bake.

<img width="2969" height="1667" alt="Image" src="https://github.com/user-attachments/assets/d400d0a3-a1b3-44b7-916f-0378dec39e94" />

**Comment by guycalledfrank** [comment]
Tried all kinds of light combinations but couldn't get the same result. Can you send me a package with these rooms + lights?
